### ISSUE TITLE
Implement basic logic within .act files

### ISSUE THREAD

**Issue by eeijlar** [mlieshoff/exactor#34]
Often you would like to skip certain tests based on certain conditions. It would be nice if the following was possible in Exactor:

```
# Set var
SetVar var [RunVerbWhichReturnsVar]
 
# Set a property
SetProperty isAboveCertainValue "{[var] >  2.1}"
 
# Run next verb based on property
Guard [isAboveValue] RunSomethingElse

```
This would all the return value of a verb to be assigned to a variable.
Guard would execute a simple if statement based on the returned value.

**Comment by mlieshoff** [comment]
Hi eeijlar. If i understand correctly, we need to implement a guard as command. Placed it into an .act file, it will ignore the test or not, based on condition. The number of ignores will be countered. 
What do you mean with "RunVerbWhichReturnsVar", is it a command? And same for "RunSomethingElse".

Best regards

**Comment by eeijlar** [comment]
Yes, correct `Guard ` would be a new command and would run based on condition. It should just skip that test, without throwing an error. 

Not sure what you mean by: 'The number of ignores will be countered'

`RunVerbWhichReturnsVar` could be any verb. The verb would return a value. It could be a String, int etc. `SetVar` would allow you to assign the return value of the verb to a variable, `var` in this case. `SetVar` and `SetProperty` are similar but properties are more like constants, and variables could change.

`RunSomethingElse` is just any verb.

The basic idea is to implement a Guard command which allows you to run a command based on a certain condition - a simplified if statement essentially.

The second part is the ability to assign the return value of verbs to a variable, and use that variable later in the script.

Hope that's clear
John

**Comment by mlieshoff** [comment]
Hi John, okay it's clear except the verbs. I understand verbs are functions that return any kind of values? As it is, how we can differ verbs from variable usage in []? [xyz] means verb or variable name?

**Comment by eeijlar** [comment]
The verbs are just dummy names for verbs, just to give a more concrete example:

`# Set var
SetVar androidVersion [GetAndroidVersion] <- This is a verb which can return a value

# Set a property
SetProperty isGreaterThan2dot4 "{[androidVersion] >  2.4}"

# Run next verb based on property
Guard [isGreaterThan2dot4] Run24TestSuite`

GetAndroidVersion is a verb which returns the android version. Based on that I can 'choose' whether to run another set of tests or not. Hope that makes sense.

I think if it's in [], then it is expecting a verb which returns a value, if not it is expecting a string or int etc.

**Comment by mlieshoff** [comment]
Hi eeijlar

a first shot (the actual trunk) is ready for trying out. You can register verbs like:

`
        public class GetAndroidVersion implements Verb...
`

`
        Runner runner = new Runner("directory");
`
`
        runner.registerVerbs(new GetAndroidVersion());
`

`
SetVar androidVersion [GetAndroidVersion]
`

`
SetProperty isGreaterThan2dot4 "{[androidVersion] > 2.4}"
`

`
public class Run24TestSuite extends Command...
`

`
Guard [isGreaterThan2dot4] Run24TestSuite
`

Best regards

**Comment by mlieshoff** [comment]
https://bintray.com/mlieshoff/maven/exactor/1.1.30.081216

**Comment by eeijlar** [comment]
Wow! That's great ... will be a few weeks before I can try it out, will let you know

**Comment by eeijlar** [comment]
This doesn't seem to have been added to the main maven repository. Our local artifacts server doesn't have it. Is there somewhere I can download it from?

**Comment by mlieshoff** [comment]
Hi eeijlar, try that one please: https://bintray.com/mlieshoff/maven/exactor/1.1.30.111316

**Comment by eeijlar** [comment]
Thanks mlieshoff, that works but only for core, the other jars are missing:

```
[WARNING] The POM for exactor.swing:exactor-swing:jar:1.1.30.111316 is missing, no dependency information available
[WARNING] The POM for exactor.web:exactor-web:jar:1.1.30.111316 is missing, no dependency information available

```

**Comment by mlieshoff** [comment]
This should solve the problems:

https://bintray.com/mlieshoff/maven/exactor/1.1.30.170126

**Comment by eeijlar** [comment]
Sorry, no - it doesn't work - can't find any of the pom files now:

```
[WARNING] The POM for exactor.core:exactor-core:jar:1.1.30.170126 is missing, no dependency information available
[WARNING] The POM for exactor.swing:exactor-swing:jar:1.1.30.170126 is missing, no dependency information available
[WARNING] The POM for exactor.web:exactor-web:jar:1.1.30.170126 is missing, no dependency information available
```

**Comment by mlieshoff** [comment]
Is it working if you only depend on modules you need:

as example:

`
<dependency> 
    <groupId>exactor.core</groupId> 
    <artifactId>exactor-core</artifactId> 
    <version>1.1.30.111316</version> 
</dependency>
`

**Comment by mlieshoff** [comment]
sorry, that version i mean: 1.1.30.170126

**Comment by eeijlar** [comment]
No, that doesn't work either:

`[WARNING] The POM for exactor.core:exactor-core:jar:1.1.30.170126 is missing, no dependency information available`

Where does the POM live - don't see it in the central maven repo, which our artifacts server syncs with.

**Comment by eeijlar** [comment]
Small suggestion around the implementation... Guard should accept null/not exists. So if a property or variable doesn't exist or is null it will skip the verb.

Also, for the syntax can we change to the following:
Guard [**<**isGreaterThan2dot4**>**] Run24TestSuite

The <> brackets indicate that a test is being performed and distinguishes it from a regular property.

**Comment by mlieshoff** [comment]
Thank you, for the feedback. Can you try to use the jcenter repository?

**Comment by eeijlar** [comment]
Apologies, the syntax I used above doesn't make sense, it should be:
Guard <[isAboveValue]> RunSomethingElse

Note the changed brackets - sorry, I made the previous comment from memory, and it was incorrect.
